ScreenControl_Web/src/Connection/MessageReceiver.ts

```diff
@@ -0,0 +1,137 @@
+import {
+    WebSocketResponse,
+    ResponseCallback
+} from './ScreenControlServiceTypes';
+import {
+    ClientInputAction, ConvertInputAction, InputType, WebsocketInputAction
+} from '../ScreenControlTypes';
+import { ConnectionManager } from './ConnectionManager';
+
+// Class: MessageReceiver
+// Handles the receiving of messages from the Service in an ordered manner.
+// Distributes the results of the messages to the respective managers.
+export class MessageReceiver {
+    // Group: Variables
+
+    // Variable: callbackClearTimer
+    // The amount of time between checks of <responseCallbacks> to eliminate expired
+    // <ResponseCallbacks>. Used in <ClearUnresponsiveCallbacks>.
+    callbackClearTimer: number = 300;
+
+    // Variable: actionCullToCount
+    // How many non-essential <ClientInputActions> should the <actionQueue> be trimmed *to* per
+    // frame. This is used to ensure the Client can keep up with the Events sent over the
+    // WebSocket.
+    actionCullToCount: number = 2;
+
+    // Variable: actionQueue
+    // A queue of <ClientInputActions> that have been received from the Service.
+    actionQueue: Array<WebsocketInputAction> = [];
+
+    // Variable: responseQueue
+    // A queue of <WebSocketResponses> that have been received from the Service.
+    responseQueue: Array<WebSocketResponse> = [];
+
+    // Variable: responseCallbacks
+    // A dictionary of unique request IDs and <ResponseCallbacks> that represent requests that are awaiting response from the Service.
+    responseCallbacks: { [id: string]: ResponseCallback; } = {};
+
+    callbackClearInterval: number;
+
+    updateInterval: number;
+
+    // Group: Functions
+    // Function: Start
+    // Unity's initialization function. Used to begin the <ClearUnresponsiveCallbacks> coroutine.
+    constructor() {
+        this.callbackClearInterval = setInterval(
+            this.ClearUnresponsivePromises as TimerHandler,
+            this.callbackClearTimer);
+
+        this.updateInterval = setInterval(
+            this.Update.bind(this) as TimerHandler,
+            (1 / 60) * 1000);
```

Comment:
Should either this be defined like callbackClearTimer OR callbackClearTimer  be removed and hard-coded into the callbackClearInterval construction? I am happy either way but we should pick one.